### PR TITLE
Enable building opening times

### PR DESCRIPTION
**Pull request by Siilwyn**
Move the opening hours from static data so we can update the data
separately from the website build.

To test locally trigger the api endpoint to fill in the data, should then show up in the frontend: http://localhost:3000/api/consume-opening-hours?secret=dev

**Todo**:
- [ ] Simplify exchange > supabase transform code
- [x] Add CRON to delete old data
- [x] Filter by opening times (per building)
- [ ] Show possible errors in notification

**Todo after merging**:
- [ ] Remove environment variable to hide opening hours
- [ ] Add CRON to update data:
```sql
select
  cron.schedule (
    'delete-old-cron-logs',
    '@daily',
    $$
    select
      net.http_get(
          url:='https://spacefinder.tudelft.nl/api/consume-opening-hours?secret=XXX'
      ) as request_id;
    ;
    $$
  );
```

### CODE REVIEW

File: src/components/SpaceCard/SpaceCard.vue [hunk 1]
```diff
@@ -69,19 +69,6 @@
...
         />
-
-        <CardStatus
-          :opening-hours="space.openingHours"
-          class="space-card__open-status"
-          :class="{ 'space-card__open-status--visible': !hideOpeningHours }"
-        />
-
-        <OpeningHours
-          v-if="!hideOpeningHours"
```
Comment (jbmoelker) [inline src/components/SpaceCard/SpaceCard.vue:80]: Remove the concept of `hideOpeningHours` from the code base. No longer need it or used.

**Review by jbmoelker — APPROVED** [review]

File: src/composables/useSpacefinderSupabase.ts [hunk 2]
```diff
@@ -26,8 +26,22 @@ export default function useSpacefinderSupabase() {
     return data;
   }
 
+  function getBuildings() {
+    return client
+      .from("buildings")
+      .select("number, opening_hours")
+      .then(({ data, error }) => {
+        if (error) {
```
Comment (jbmoelker) [inline src/composables/useSpacefinderSupabase.ts:34]: reflect error in UI?

File: src/stores/map.ts [hunk 3]
```diff
@@ -38,7 +38,7 @@ export const useMapStore = defineStore("map", () => {
           buildingAbbreviation: space.building.abbreviation,
           buildingNumber: space.building.number,
           buildingOccupancy: space.building.occupancy,
-          isOpen: spaceIsOpen(now, space.openingHours),
+          isOpen: true,
```
Comment (jbmoelker) [inline src/stores/map.ts:41]: have another look at this one :)

**Review by jbmoelker — APPROVED** [review]

File: src/composables/useSpacefinderSupabase.ts (hunk 2 above)
Comment (Siilwyn) [reply src/composables/useSpacefinderSupabase.ts:34]: There is a NotificationBar component but no straightforward way to pass messages to it from a composable, something for a ticket maybe? A general queue of messages to show could be useful for feedback (errors and info).

File: src/server/api/consume-opening-hours.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    .from("buildings")
+    .upsert(
+      buildingOpeningHoursResult.map((building) => ({
+        number: building.exchangeBuildingId.replace(/\D/g, ""),
+        opening_hours: building.openingHours,
+      }))
+    );
+
+  if (error) {
+    console.error("Failed to upsert buildings to Supabase", error);
```
[the remark this answers is not in the document]
Comment (Siilwyn) [reply src/server/api/consume-opening-hours.ts:50]: Done in 7af8b1d

File: src/stores/map.ts (hunk 3 above)
Comment (Siilwyn) [reply src/stores/map.ts]: Removed in f0e9a47 it looks like this state is unused? The filter and such still work without it.

**Pull request merged**